### PR TITLE
Chaoslabs proxy

### PR DESCRIPTION
**Pull request by valiafetisov**
Contributes to #83 

Testing is explained in the [new readme](https://github.com/sidestream-tech/unified-auctions-ui/blob/802615f224026e0e79a8bbe46bfa45c320a4e8f5/chaos-proxy/README.md).

Please note that latest `CRVV1ETHSTETH` simulation will not fetch anything on this branch, because `CRVV1ETHSTETH` will be implemented in the #156

### CODE REVIEW

File: chaos-proxy/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Chaos Simulation Reverse Proxy
+
+The package to access chaoslabs simulations locally. Forked from the [ChaosLabsInc/chaos-server-proxy](https://github.com/ChaosLabsInc/chaos-server-proxy).
+
+## Functionality
+
+1. Spin up a remote in-the-cloud simulation based on given enviourment variables
```
Comment (benediktkaiser) [inline chaos-proxy/README.md:7]:
```suggestion
1. Spin up a remote in-the-cloud simulation based on given environment variables
```

File: chaos-proxy/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+2. Start a reverse proxy to the simulation RPC node and serve it on localhost
+
+## Environment variables
+
+- `CHAOSLABS_ACCESS_TOKEN`: (required) access topken (acquired from the chaoslabs)
```
Comment (benediktkaiser) [inline chaos-proxy/README.md:12]:
```suggestion
- `CHAOSLABS_ACCESS_TOKEN`: (required) access token (acquired from the chaoslabs)
```

File: chaos-proxy/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  create new project -> settings -> keys). Note: this project can not be restricted by the origin.
+
+### How to start
+
+1. Create `chaos-proxy/.env` file with required env variables (see explanation above)
+    - If you have access to the chamber secrets, you can instead execute this command inside this folder:
+    ```
+    secret export auction-ui/localhost/chaos-proxy --format dotenv > ./.env
+    ```
+2. Install dependecies with `npm i`
```
Comment (benediktkaiser) [inline chaos-proxy/README.md:24]:
```
2. Install dependencies with `npm i`
```

File: chaos-proxy/package.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "dydx-cli",
```
Comment (benediktkaiser) [inline chaos-proxy/package.json:2]: Just out of curiosity what does `dydx` stand for?

File: chaos-proxy/src/api.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+const fetchSimulationUrl = async function (accessToken: string, instanceId: string): Promise<string | undefined> {
+    const response = await executeAPIrequest(accessToken, '/simulation_details', { id: instanceId });
+    if (response.status === 404) {
+        return undefined;
+    }
+    return response.data?.nodeUrl;
+};
+
+const waitForSimluationUrl = async function (accessToken: string, instanceId: string): Promise<string | undefined> {
```
Comment (benediktkaiser) [inline chaos-proxy/src/api.ts:40]:
```suggestion
const waitForSimulationUrl = async function (accessToken: string, instanceId: string): Promise<string | undefined> {
```

File: chaos-proxy/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+(async function () {
+    if (!CHAOSLABS_ACCESS_TOKEN) {
+        throw new Error('CHAOSLABS_ACCESS_TOKEN is not set');
+    }
+    if (!CHAOSLABS_SIMULATIONS) {
+        throw new Error('CHAOSLABS_SIMULATIONS is not set');
+    }
+    const simulationId = await chooseSimulationId(CHAOSLABS_SIMULATIONS);
+    const targetURL = await getSimulationUrl(CHAOSLABS_ACCESS_TOKEN, simulationId);
+    startProxy(targetURL, PORT);
```
Comment (benediktkaiser) [inline chaos-proxy/src/index.ts:19]: Doesn't this need to be awaited as it is a async function?

**Review by benediktkaiser — CHANGES_REQUESTED** [review]

File: chaos-proxy/package.json (hunk 4 above)
Comment (valiafetisov) [reply chaos-proxy/package.json]: It wasn't changed from the original code I've copied over (the link to the original is in the readme)

**Review by LukSteib — CHANGES_REQUESTED** [review]
Is there a chance for me to import `DAI` as token to Metamask on the localhost? Would help in order to improve testing quality of execution. 

When trying to import the mainnet token address metamask throws an error. (see below)

image

---

- [ ] The 5 ETH-A simulation provides not 5 ETH-A auctions but ~30. So we might think about renaming that simulation
- [ ] Additionally the wallet was not funded with ETH for this 5 ETH-A simulation

---

- [ ] The 2 ETH-A simulation provides only 1 ETH-A auction. We might rename it.

**Comment by valiafetisov** [comment]
- All the simulation ids and their names are stored in the secrets. I renamed it without modifying the code
- The amount of auctions depends of the market conditions, so yeah I will just call them `Some ETH-A` and `Many ETH-A` since we weren't provided with actual explanations what is the difference between them

> Additionally the wallet was not funded with ETH for this 5 ETH-A simulation

You can try to report this to chaoslabs. This PR only implements a bridge between localhost and their cloud service + a nice UI to store and select simulations

> Is there a chance for me to import DAI as token to Metamask on the localhost?

Hmm, good question, but I couldn't google it right away. From our side, we should just add button to open "Manage DAI in VAT" popup to the header.

**Comment by LukSteib** [comment]
> You can try to report this to chaoslabs.

Will do 

> From our side, we should just add button to open "Manage DAI in VAT" popup to the header.

Sounds like a good workaround

**Comment by valiafetisov** [comment]
Then, would you already approve this PR? 🙂

**Review by LukSteib — APPROVED** [review]

**Pull request merged**